File: Ejercicio_03/Domain/Interfaces/Persona/IPersonaRepository.cs
using Domain.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Interfaces 
{
    public interface IPersonaRepository
    {
		List<Persona> getPersonas();
		Persona getPersona(int id);
        int deletePersona(int id);
        int updatePersona(int id, Persona persona);
        int addPersona(Persona persona);
    }
}

File: Angular/CRUD/src/domain/interfaces/IPersonaRepository.ts
import { Persona } from "../entities/Persona"
import { PersonaDTO } from "../dtos/PersonaDTO"

export interface IPersonaRepository {
    getPersonas(): Promise<Array<PersonaDTO>>
	getPersona(id: number): Promise<PersonaDTO>
	updatePersona(id: number, persona: Persona): Promise<number>
	addPersona(persona: Persona): Promise<number>
	deletePersona(id: number): Promise<number>
}